Sources/GRPC/ClientConnection.swift

```diff
@@ -112,9 +110,23 @@ public class ClientConnection {
     self.configuration = configuration
     self.scheme = configuration.tls == nil ? "http" : "https"
     self.authority = configuration.tls?.hostnameOverride ?? configuration.target.host
+
+    let monitor = ConnectivityStateMonitor(
+      delegate: configuration.connectivityStateDelegate,
+      queue: configuration.connectivityStateDelegateQueue
+    )
+
+    self.connectivity = monitor
     self.connectionManager = ConnectionManager(
       configuration: configuration,
-      logger: configuration.backgroundActivityLogger
+      logger: configuration.backgroundActivityLogger,
+      onConnectivityStateChange: { manager, _, newState in
+        let logger = manager.logger
+        monitor.updateState(to: newState, logger: logger)
+      },
+      onConnectionIsQuiescing: { _ in
+        monitor.beginQuiescing()
+      }
```

Comment:
Are we sure we want to have the communication `ConnectionManager` -> `ConnectivityStateMonitor` closure based? Is this the right tradeoff vs. a protocol here? This is not really a change suggestion, more a question how we deal with these type of situations here.

Reply:
Changed to a protocol (two protocols actually, one for connectivity another for http/2 things).

> Using a protocol existential would be cheaper, if we can do it.

Because we save on allocating closure contexts?

Reply:
Yes: protocol existentials only require heap allocation if you're stuffing a closure larger than 24 bytes into them, otherwise they're stack allocated. You can't optimise through them, but often you can't do that with a closure _anyway_.